ingestion-edge/ingestion_edge/publish.py

```diff
@@ -68,42 +70,78 @@
         return await async_wrap(future)
 
 
-async def flush(request: Request, q: SQLiteAckQueue) -> response.HTTPResponse:
-    """Flush messages from the local queue to pubsub.
+def _set_status(q, message, future):
+    """Set message status from future."""
+    try:
+        # detect exception
+        future.result()
+    except:  # noqa: E722
+        # message was not delivered
+        q.nack(message)
+        # raise from bare except
+        raise
+    else:
+        # message delivered
+        q.ack(message)
+
 
-    Call periodically on each docker container to ensure reliable delivery.
+async def flush(q: SQLiteAckQueue):
+    """Periodically flush messages from the local queue to pubsub.
 
-    Access to this endpoint should be restricted.
+    Attempts to flush the entire queue every 60 seconds.
+
+    Must not exit unless an interrupt is received.
     """
-    if q.size == 0:
-        # early return for empty queue
-        return response.text("", HTTP_STATUS.NO_CONTENT)
-    result = {DONE: 0, PENDING: 0}
-    # while queue has messages ready to process
-    while q.size > 0:
+    # run forever
+    while True:
         try:
-            message = q.get(block=False)
-        except Empty:
+            # flush until the queue is empty or a publish fails
+            while True:
+                pending, size = [], 0
+                for _ in range(1000):
+                    if size > 5000000:
```

Comment:
Where do these literals 1000 and 5000000 come from? They should probably be tunable configuration parameters or named constants.

Reply:
1000 is the max batch size for the pubsub api , and 5000000 was chosen somewhat arbitrarily to keep the batch from running over the 10MB limit without serializing to protobuf because we can't reuse the serialized message in the client.

that said, they should totally be tunable, and be tuned based on load testing results. will fix.